embed/config.go

```diff
@@ -381,8 +381,8 @@ func (cfg *Config) UpdateDefaultClusterFromName(defaultInitialCluster string) {
 		// if client-listen-url is 0.0.0.0, just use detected default host
 		// otherwise, rewrite advertise-client-url with localhost
 		if ip != "0.0.0.0" {
-			_, acPort, _ := net.SplitHostPort(cfg.ACUrls[0].Host)
-			cfg.ACUrls[0] = url.URL{Scheme: cfg.ACUrls[0].Scheme, Host: fmt.Sprintf("localhost:%s", acPort)}
+			ahost, acPort, _ := net.SplitHostPort(cfg.ACUrls[0].Host)
```

Comment:
This seems wrong. Why do you parse the host and port and set the exactly same host and port back?